### DISCUSSION TITLE
[Report bugs] The bug in read/convert data from grb2 to nc through Xarray-python

### DISCUSSION THREAD

**Discussion by ruhuihuang** [pydata/xarray#6643] [Q&A]
Hi there:

This bug occurs when there is no variable name and multiple variables are mixed together in file of grb format. 

Then if you use  
`data = xr.open_dataset( 'filename.grb2',engine='cfgrib', backend_kwargs={'filter_by_keys': {'typeOfLevel': '.......'}})` or other kind of codes of Xarray to read grb2 data or to convert grb2 data to nc data, although Xarray can read/convert it, you will get incorrect data. 

You can have a try using the data from [](https://climate.mri-jma.go.jp/pub/ocean/ts/v7.3.1/temp/) to have a try, also for convenience you can use the data in the attachment to have a try. 

On the other hand, even you use `data=pygrib.open('filename.grb2')`, then you will find it hard to get anything you want since it has no name.

Best regards

Ruhui Huang
Xiamen University
ruhui@stu.xmu.edu.cn

Attachment:
[temp.1955.grb2.zip](https://github.com/pydata/xarray/files/8784748/temp.1955.grb2.zip)

**Comment by raybellwaves** [comment]
You may be seeing https://github.com/ecmwf/cfgrib/issues/245
